From 1fd1806ace2728fb37c2e6aa8cdbe390ef36e850 Mon Sep 17 00:00:00 2001
From: Samreen Nilofer <samreen.nilofer@intel.com>
Date: Thu, 12 Feb 2015 10:05:28 +0530
Subject: [PATCH 19/24] ASoC: intel: platform: Add PCM3 and PCM4 OUT path.

Issue: GMINL-4319

Adding two new pipes from SBA to allow multichannel capture.
These two pipes are combined via an interleaver to give a
4-channel capture frontend.

Change-Id: I61820f0ce5aa457356ab11de3f28d223bcec2750
Signed-off-by: Praveen Diwakar <praveen.diwakar@intel.com>
Signed-off-by: Samreen Nilofer <samreen.nilofer@intel.com>
---
 sound/soc/intel/platform-libs/atom_controls.h    | 5 ++++-
 sound/soc/intel/platform-libs/atom_pipes.h       | 6 ++++++
 sound/soc/intel/platform-libs/controls_v2_dpcm.c | 3 +++
 3 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/sound/soc/intel/platform-libs/atom_controls.h b/sound/soc/intel/platform-libs/atom_controls.h
index 1d813b7..9c272d4 100644
--- a/sound/soc/intel/platform-libs/atom_controls.h
+++ b/sound/soc/intel/platform-libs/atom_controls.h
@@ -58,7 +58,10 @@
 #define SST_MIX_MEDIA0		SST_MIX(19)
 #define SST_MIX_MEDIA1		SST_MIX(20)
 
-#define SST_NUM_MIX		(SST_MIX_MEDIA1 + 1)
+#define SST_MIX_PCM3		SST_MIX(21)
+#define SST_MIX_PCM4		SST_MIX(22)
+
+#define SST_NUM_MIX		(SST_MIX_PCM4 + 1)
 
 #define SST_MIX_SWITCH		(SST_NUM_MIX + 1)
 #define SST_OUT_SWITCH		(SST_NUM_MIX + 2)
diff --git a/sound/soc/intel/platform-libs/atom_pipes.h b/sound/soc/intel/platform-libs/atom_pipes.h
index 9fe301b..4baaa5a 100644
--- a/sound/soc/intel/platform-libs/atom_pipes.h
+++ b/sound/soc/intel/platform-libs/atom_pipes.h
@@ -53,6 +53,8 @@ enum sst_dfw_path_index {
 	SST_DFW_PATH_INDEX_PCM0_OUT	      = (0x0D << SST_DFW_PATH_ID_SHIFT),
 	SST_DFW_PATH_INDEX_PCM1_OUT	      = (0x0E << SST_DFW_PATH_ID_SHIFT),
 	SST_DFW_PATH_INDEX_PCM2_OUT	      = (0x0F << SST_DFW_PATH_ID_SHIFT),
+	SST_DFW_PATH_INDEX_PCM3_OUT	      = (0x10 << SST_DFW_PATH_ID_SHIFT),
+	SST_DFW_PATH_INDEX_PCM4_OUT	      = (0x11 << SST_DFW_PATH_ID_SHIFT),
 	SST_DFW_PATH_INDEX_AWARE_OUT	      = (0x12 << SST_DFW_PATH_ID_SHIFT),
 	SST_DFW_PATH_INDEX_VAD_OUT	      = (0x13 << SST_DFW_PATH_ID_SHIFT),
 
@@ -203,6 +205,10 @@ enum sst_dfw_swm_outputs {
 						SST_DFW_DEFAULT_CELL_NBR),
 	SST_DFW_SWM_OUT_PCM2		= (SST_DFW_PATH_INDEX_PCM2_OUT	      |
 						SST_DFW_DEFAULT_CELL_NBR),
+	SST_DFW_SWM_OUT_PCM3		= (SST_DFW_PATH_INDEX_PCM3_OUT	      |
+						SST_DFW_DEFAULT_CELL_NBR),
+	SST_DFW_SWM_OUT_PCM4		= (SST_DFW_PATH_INDEX_PCM4_OUT	      |
+						SST_DFW_DEFAULT_CELL_NBR),
 	SST_DFW_SWM_OUT_AWARE		= (SST_DFW_PATH_INDEX_AWARE_OUT	      |
 						SST_DFW_DEFAULT_CELL_NBR),
 	SST_DFW_SWM_OUT_VAD		= (SST_DFW_PATH_INDEX_VAD_OUT	      |
diff --git a/sound/soc/intel/platform-libs/controls_v2_dpcm.c b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
index 9453864..6634e0b 100644
--- a/sound/soc/intel/platform-libs/controls_v2_dpcm.c
+++ b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
@@ -1851,6 +1851,9 @@ static const struct snd_kcontrol_new sst_probe_controls[] = {
 		sst_gain_tlv_common, gain_var)
 
 
+/* Note: NUM_GAIN is common between DFW and kernel,
+ * keep note while modifying this
+ */
 #define SST_NUM_GAINS 36
 static struct sst_gain_value sst_gains[SST_NUM_GAINS];
 
-- 
1.9.1

